switchapi/switch_neighbor_int.h

```diff
@@ -22,10 +22,16 @@
 extern "C" {
 #endif
 
-#include "switch_pd_routing.h"
 #include "switch_rif.h"
 #include "switch_types_int.h"
 
+// TODO Nupur: Check with Derek
```

Comment:
Just `#include "switch_pd_routing.h"`.

If you look at `switchapi/switch_pd_routing.h`, you'll see that it's a wrapper file that chooses between the dpdk and es2k implementations of `switch_pd_routing.h`.

We have a choice then between (1) a quick-and-dirty fix and (2) a fix that moves us in a more general direction.

The quick-and-dirty fix is to change the `"include es2k/switch_pd_routing.h"` in the wrapper file to `"include switchapi/es2k/lnw_v3/switch_pd_routing.h"`.

The more general fix is to combine the lnw_v2 and lnw_v3 versions of `switch_pd_routing.h` into a single file in the `es2k` folder.

- Change `#include "switchapi/es2k/lnw_v3/switch_pd_p4_name_routing.h"` back to `switch_pd_p4_name_routing.h`.
-  Provide function prototypes for _both_ `switch_pd_neighbor_table_entry()` and `switch_pd_ecmp_nexthop_table_entry()`. Put a comment on each one saying which LNW version it applies to.

This is the (almost) ideal solution. We have created a single interface to both es2k implementations of `switch_pd_routing.c`. The ideal solution would be to have a single function with two different implementations (abstraction is a good way to hide implementation details).
 
That leaves us with what to do about the `switch_pd_p4_name_routing.h`. My suggestion would be to:
-  Create an `es2k/switch_pd_p4_name_routing.h` wrapper file along the lines of `switchapi/switch_pd_routing.h`. 
- Have it include `switchapi/es2k/lnw_v3/switch_pd_p4_name_routing.h` for now.
- Once we have conditionals for the LNW versions, we can change it to include the correct version-specific header file.

Reply:
Addressed